coalib/collecting/Collectors.py

```diff
@@ -126,11 +126,13 @@ def list_glob_results(values=None):
 
 
 def collect_files(file_paths, log_printer=None, ignored_file_paths=None,
-                  limit_file_paths=None, section_name=''):
+                  limit_file_paths=None, section_name='',
+                  line_num=[]):
     """
     Evaluate globs in file paths and return all matching files
 
     :param file_paths:         File path or list of such that can include globs
+    :param line_num:           List of line numbers which correspond to files
```

Comment:
remove this parameter, and all changes to coalib/processing and coalib/collecting , etc.

we do not want to be passing around line numbers.   We pass around `SourcePosition` objects, which contain the filename and the line number together as a single object.

Your PR should not try to implement the better error messages everywhere.  Focus on `Setting.origin` supporting `SourcePosition` in a backwards compatible way.

Create unit tests which demonstrate the the Setting.origin are being set with the correct values.